src/index.js

```diff
@@ -79,6 +81,12 @@ export default class SuperImage extends React.Component {
         props[key] = this.props[key];
       });
 
+    // If alt is empty, only 'img' is specified as role
+    // @see https://www.w3.org/TR/html-aria/#img-alt
+    if (props.alt && props.role) {
+      props.role = 'img';
+    }
```

Comment:
@tommy-san @masuP9 

In this case `null` may be the most reasonable ...?
What do you think of this?

Reply:
@tsuyoshiwada

Me too.
I think that `null` or remove role property is good.

>Implicit ARIA semantics - SHOULD NOT be used

ref https://www.w3.org/TR/html-aria/#docconformance

@masuP9 

What do you think?

Reply:
Just Idea. (言ってみたかった） `img` elemenet don't need `role='img'`. 👇 

```JSX
const { role } = this.props;
return (
  <img role={role && role !== 'img' ? role : null} />
);
```

Reply:
@masuP9 

With reference to the code you presented, I thought that the code below is good :bulb:

```javascript
if (props.alt && props.role !== 'img') {
  props.role = 'img';
}
```

Reply:
In this code, finally `img` element has `role="img"` when has alt text and role defined other than img (eg. presentation).

I think its necessary to split the process in case of img element and fallback case.

Reply:
@masuP9 I was writing a big mistake ... :crying_cat_face:

> In this code, finally img element has role="img" when has alt text and role defined other than img (eg. presentation).

Yes. You're absolutely right ! (Test case was also wrong...)

I think the following code is correct, what do you think??

**Implementation:**

```javascript
// for <img /> props
if (props.alt && props.role !== 'img') {
  props.role = null;
}
```

**TestCase:**

```javascript
const superImage = TestUtils.renderIntoDocument(<SuperImage src="" role="presentation" alt="some text" />);
const node = TestUtils.findRenderedDOMComponentWithTag(superImage, 'img');

assert(!node.hasAttribute('role'));
```

Reply:
@masuP9 ping :wink:

Reply:
@tsuyoshiwada pong 🏓 

LGTM!

Reply:
@masuP9 Thx :sparkles:  Then I will send a new PR for fix :+1: